apps/emqx_rule_engine/test/emqx_rule_funcs_SUITE.erl

```diff
@@ -1003,6 +1003,24 @@ prop_format_date_fun() ->
                         )
                     ]
             )
+    ),
+    %% When no offset is specified, the offset should be taken from the formatted time string
+    ArgsNoOffset = [<<"second">>, <<"%y-%m-%d-%H:%M:%S%Z">>],
+    ArgsOffset = [<<"second">>, <<"+08:00">>, <<"%y-%m-%d-%H:%M:%S%Z">>],
+    ?FORALL(
+        S,
+        erlang:system_time(second),
```

Comment:
I think this is called only once for all tests.  I tried locally with `nanosecond` and printing the value, it's always the same.

To achieve the desired generation of values, I think one way is to do:

```suggestion
        ?LAZY(erlang:system_time(second)),
```

Not sure if the most correct way to produce values, but does produce different values.

Also, note that the generation is very fast: it produces several values in a second.  So, even with `?LAZY`, this will basically always return the same value.

Perhaps we could instead define a range of epoch values?  Like

```suggestion
        range(0, 4000000000),
```

Reply:
Thanks, I think this problem existed in the other test cases in the same function as well. Will try to fix them too.

Reply:
@thalesmg thanks for spotting this. I have updated the test case in the latest commit.